core/wrapTableIterators.ml

```diff
@@ -0,0 +1,61 @@
+open Utility
+open CommonTypes
+open Sugartypes
+(*
+ * Ensures that table iterators are wrapped in a query block:
+ *
+ * for (x <-- y) M
+ *  --->
+ * query { for (x <-- y) M }
+ *
+ * Note that this is safe if we're in another query block:
+ *
+ * query { for (x <-- y) M }
+ *  --->
+ * query { query { for (x <-- y) M } }
+ *
+ * Since nested query blocks are removed by the normaliser.
+*)
```

Comment:
In fact, unlike when this pass was done in the typechecker, the new pass doesn't need to operate inside `query` blocks since it's redundant -- so this comment is outdated. Thanks for picking it out; I'll update it.